torch/testing/_internal/common_utils.py

```diff
@@ -3250,12 +3251,13 @@ def load_tests(loader, tests, pattern):
     set_running_script_path()
     test_suite = unittest.TestSuite()
     for test_group in tests:
-        for test in test_group:
-            check_test_defined_in_running_script(test)
-            test_suite.addTest(test)
+        if not DISABLE_RUNNING_SCRIPT_CHK:
+            for test in test_group:
+                check_test_defined_in_running_script(test)
+        if test_group._tests:
```

Comment:
Do you need this check `if test_group._tests`? I assume that `test_suite.addTest(test_group)` alone would work just fine.

Reply:
> Thank you for your contribution!
> 
> Overall, the change looks good to me. I have double check several unittest XML outputs that we use and found no issue when the testsuites are not flatten. Please also note that we are slowing moving toward pytest, so load_tests might become less relevant in the future.
> 
> I'll approve once all tests pass.

Great to see the move to pytest! (I've found it preferable for numerous reasons in the past)

Reply:
> Do you need this check `if test_group._tests`? I assume that `test_suite.addTest(test_group)` alone would work just fine.

I agree this check shouldn't be strictly necessary, but I was thinking it's a bit cleaner to only return non-empty test suites for downstream consumption.  For example, `unittest` discovery  will sometimes return `TestSuite`s [without any tests](https://github.com/python/cpython/blob/96f8d3619d839266491b722b943de65892bb0e81/Lib/unittest/loader.py#L121-L124)
Concretely:
```python
# examining load_tests test_suite return using the test_*cuda*.py pattern
# with check
[len(suite._tests) for suite in test_suite._tests]
[144, 26]
# without check
[len(suite._tests) for suite in test_suite._tests]
[0, 144, 26]
```
I'm happy to remove the check if you prefer that approach though, just lmk. Thanks again for the PR review and all your contributions!